File: java_programs/stack.java
public class stack{
	public static class Node{
		int val;
		Node next;
	}

	public static class stk{
		
		Node top = null;
		Node bottom = null;
		int size = 0;
		
		public void push(int val){
			Node nn = new Node();
			nn.val = val;
			nn.next = null;
			if(this.top != null)
				nn.next = top;
			else
				this.bottom = nn;
			this.top = nn;
		}
		public int pop(){
			Node rv = this.top;
			this.top = this.top.next;
			rv.next = null;
			return rv.val;
		}
		public void display(){
			Node p = this.top;
			while(p!=null){
				System.out.println(p.val);
				p = p.next;
			}
		}
	}

	public static void main(String[] args) {
		stk s = new stk();	
		s.push(10);
		s.push(20);
		s.push(30);
		s.display();
		System.out.println("-----------------");
		System.out.println(s.pop());
		System.out.println("-----------------");
		s.display();
	}
}

File: python_programs/stack.py
class Node:
	def __init__(self , val , next):
		self.val = val
		self.next = next

class stack:
	def __init__(self):
		self.head = Node(None , None)
		self.bottom = Node(None , None)
		self.size = 0

	def push(self,val):
		nn = Node(val , None)
		if self.head.val != None:
			nn.next = self.head
		else:
			self.bottom = nn
		self.head = nn
		self.size+=1

	def pop(self):
		rv = self.head
		self.head = self.head.next
		rv.next = None
		self.size -= 1
		return rv.val

	def display(self):
		p = self.head
		while p!= None:
			print(p.val)
			p = p.next

s = stack()
s.push(10)
s.push(20)
s.push(30)
s.display()
print('--------------')
print(s.pop())
print('--------------')
s.display()